lib/adb.js

```diff
@@ -46,10 +47,16 @@ var ADB = function (opts) {
   if (opts.remoteAdbPort) {
     this.adb.defaultArgs.push("-P", opts.remoteAdbPort);
   }
+  var initError = null;
+  async.series([
+    this._checkAdbPresent.bind(this)
+  ], function(err) {
+    initError = err;
+  });
+  if (initError) throw initError;
```

Comment:
there's an async bug here: this line will usually execute before `this._checkAdbPresent` ever has a chance to.

we can't run async functions inside a constructor. if we want to do something like this we should create an `init` function or something which is async